### DISCUSSION TITLE
Your opinion to Max from Mojo

### DISCUSSION THREAD

**Discussion by Jumaron** [AmusementClub/vs-mlrt#87] [General]
With the release of Max from Mojo, I wonder what you think of it compared to the current fastest option TensorRT.
In there Discord they claim that it should replace other engines like TensortRT and Pytorch.
Whats your opinion?

**Comment by WolframRhodium** [comment]
Firstly, Max for NVIDIA GPUs is not available yet, but they claimed on GTC 2024 last week that the initial support will be out in 24Q2, and they shown promising matrix multiplication performance on RTX 4070 Ti compared to cuBLAS; on the other hand, TensorRT is an established framework. 

Secondly, both frameworks are written by experts in the field.
- TensorRT is free for customers because NVIDIA makes a profit from selling its hardware.
- MAX has both a developer edition and an enterprise edition, because Modular is a software company and it needs to make a profit. It is unclear to me whether there will be some features exclusive to their enterprise edition.

As a consequence, Modular may and should prioritize support for popular deep learning models like Llama 2 and Stable Diffusion, and it may not support operations used in vs-mlrt soon. For example, the latest edition (24.2.0) for CPU does not support PReLU, PixelShuffle and GridSample operations that are widely used.

Finally, it's always good to see competition. I do consider including it as another backend option. I also read all their blogs and documentation and watch their videos, and I have learned a lot from that.

**Reply by Jumaron** [reply]
It's nice to hear you're also watching the news.

This year will be eventful. I'm hyped.

Nonetheless, we will slowly need updated models, for example, Cugan and ESRGAN. They are already a bit "old" How time flies 😭

**Reply by Jumaron** [reply]
Also I hope Mojo brings out Windows support, you use Mac or Linux right?

**Reply by WolframRhodium** [reply]
Yes, I use Ubuntu and MAX is only available on it at this time.

**Reply by Jumaron** [reply]
Gigachad, let us know when you have tested MAX how it is ^^

**Reply by WolframRhodium** [reply]
Sure.
